File: 7_pyramidArray_101319.js
// My solution:
function pyramid(n){
    let answer = [];
    for (i=0; i<n; i++) {
        a=new Array(i)
        for (j = 0; j <= i; ++j) {
            a[j] = 1;
        }
        answer.push(a);
    }
    return answer;
}

pyramid(4);

// Best Practice solution from DeadIntegral:
function pyramid(n) {
    const res = [];
    for(let i = 0; i < n; i++){
      res.push([...Array(i+1)].fill(1))
    }
    return res;
  }

// Best Practice solution from SavagePixie:
const pyramid = n => Array(n).fill(1).map((x, i) => Array(i + 1).fill(1))

File: 7_pyramidArray_101319.py
# Write a function that when given a number >= 0,
# returns an Array of ascending length subarrays.
# Ex. pyramid(0) => [ ]
# Ex. pyramid(3) => [ [1], [1, 1], [1, 1, 1] ]
# NOTE: The subarrays should be filled with 1s

# This is a very ugly surefire solution:
def pyramid(n):
    block = []
    subar = []
    for num in range(n):
        subar.append(1)
        ones = subar[0:(num-1)]
        ones.append(1)
        value = num
        if num > 1:
            value = num-1
        if value > 0:
            ones.append(1)
        block.append(ones)
    return block

# Using a loop in the return statement is the key:
def pyramid(n)
    answer = []
    for i in range(1,n+1):
        answer.append(i * [1])
    return answer

# Best Practice soluion by sahglie:
def pyramid(n):
    return [[1]*x for x in range(1, n+1)]